plotly/_subplots.py

```diff
@@ -746,8 +746,12 @@ def _check_hv_spacing(dimsize, spacing, name, dimvarname, dimname):
             )
             grid_ref[r][c] = subplot_refs
 
-    _configure_shared_axes(layout, grid_ref, specs, "x", shared_xaxes, row_dir)
-    _configure_shared_axes(layout, grid_ref, specs, "y", shared_yaxes, row_dir)
+    _configure_shared_axes(layout, grid_ref, specs, "x", shared_xaxes, row_dir, False)
+    _configure_shared_axes(layout, grid_ref, specs, "y", shared_yaxes, row_dir, False)
+    if secondary_y:
+        _configure_shared_axes(
+            layout, grid_ref, specs, "y", shared_yaxes, row_dir, True
+        )
```

Comment:
@gmjw I think there's a logic issue here -- `secondary_y` is set in the loop above, which loops through all the `specs` for each subplot. So after the above loop terminates, `secondary_y` corresponds only to the value provided for `secondary_y` in the spec of the final subplot.

You can see how this causes issues if you set `'secondary_y': False` for the final subplot only; it breaks shared y-axes for every subplot in the grid.

The fix might be as simple as something like this:

```suggestion
    any_secondary_y = any([spec.get("secondary_y", False) for spec_row in specs for spec in spec_row])
    if any_secondary_y:
        _configure_shared_axes(
            layout, grid_ref, specs, "y", shared_yaxes, row_dir, True
        )
```

although I haven't fully thought through all the edge cases.

Reply:
Hi @emilykl - thanks for spotting this! I should have read through the function more carefully.

I've adjusted almost as you suggest - based on the loop above, it looks like I can rely on "secondary_y" being in `spec` provide it is not None.